tools/query_verification/src/main/java/com/google/bigquery/QueryVerifier.java

```diff
@@ -70,37 +66,49 @@ public void verifyDataFree() {
                 }
 
                 List<TableId> tableIds = QueryVerifier.getTableIdsFromDdlSchema(migratedSchema);
-                for (TableId tableId : tableIds) {
-                    Table table = bigQuery.getTable(tableId);
-                    tables.add(table);
-                }
+                tableIds.forEach(tableId -> tables.add(bigQuery.getTable(tableId)));
             }
 
             if (tables.isEmpty()) {
                 System.out.println(migratedSchema.path() + " is not correctly formatted.");
             }
         }
 
-        // Create dry-run job
-        JobInfo jobInfo = configureJob(migratedQuery.query(), true);
+        // Create dry-run jobs
+        List<JobInfo> jobInfos = getJobInfosFromQuery(migratedQuery, true);
+        ListIterator<JobInfo> jobInfoListIterator = jobInfos.listIterator();
```

Comment:
I think we should prefer a loop like ```for (int i = 0; i < jobInfos.size(); i++) { JobInfo jobInfo = jobInfos.get(i); ...```  over using listIterator, I believe it should be more clear as opposed to using the iterator methods